include/shark/Algorithms/DirectSearch/Operators/ReferenceVectorAdaptation.h

```diff
@@ -0,0 +1,58 @@
+#ifndef SHARK_ALGORITHMS_DIRECT_SEARCH_OPERATORS_REFERENCE_VECTOR_ADAPTATION_H
+#define SHARK_ALGORITHMS_DIRECT_SEARCH_OPERATORS_REFERENCE_VECTOR_ADAPTATION_H
+
+#include <shark/Algorithms/DirectSearch/Operators/Selection/ReferenceVectorGuidedSelection.h>
+
+namespace shark {
+
+struct ReferenceVectorAdaptation
+{
+	typedef shark::Individual<RealVector, RealVector> IndividualType;
```

Comment:
should be independent of the actual type of individual. better make the operators a template on the Individual. Reason is that you only use the function value of the individual, not the chromosome (last template parameter).